pkg/api/types.go

```diff
@@ -204,6 +204,8 @@ type VolumeSource struct {
 
 // Similar to VolumeSource but meant for the administrator who creates PVs.
 // Exactly one of its members must be set.
+// All PersistentVolumes are volumes, only some Volumes are persistent.  All pointers
```

Comment:
I agree.  I've been thinking about how the pointer must be in both VS and PVS.

I think I have to create the entire PersistentVolumePlugin implementation as completely separate from VolumePlugin.  This would allow the pointer to be in just PSV, leaving VS for other purposes.